src/schema/README.md

```diff
@@ -35,3 +35,165 @@ of the schema, roughly in order of importance:
+        `format` defines the format for the string at the same level as `type`.
+        Valid values for `format` include: `uri` (uniform resource
+        identifiers), `date` (date-times), `unit` (standard units),
+        `dataset_relative` (relative paths from dataset root),
+        `participant_relative` (relative paths from participant folder).
+        `enum` defines a list of valid values for the field.
+        The minimum string length (`minLength`) defaults to 1.
+        Here is an example of a field with a restricted set of possible values:
+        ```yaml
+        name: ExampleField
+        description: |
+            The description of "ExampleField".
+        type: string
+        enum:
+            - PossibleValue1
+            - PossibleValue2
+        ```
+
+    -   `integer`: If `type` is `integer`, then there MAY be any of the
+        following fields at the same level as `type`: `unit`,
+        `minimum` (inclusive minimum), `maximum` (inclusive maximum),
+        `exclusiveMinimum`, `exclusiveMaximum`.
+        Here is an example of a field with a minimum value of 0:
+        ```yaml
+        name: ExampleField
+        description: |
+            The description of "ExampleField".
+        type: integer
+        minimum: 0
+        ```
+
+    -   `number`: If `type` is `number`, then there MAY be any of the following
+        fields at the same level as `type`: `unit`,
+        `minimum` (inclusive minimum), `maximum` (inclusive maximum),
+        `exclusiveMinimum`, `exclusiveMaximum`.
+        Here is an example of a field with a minimum of 0, in units of
+        kilometers per second:
+        ```yaml
+        name: ExampleField
+        description: |
+            The description of "ExampleField".
+        type: number
+        minimum: 0
+        # k stands for "kilo", m stands for "meter", "/" means "per", and
+        # s stands for "seconds"
+        unit: km/s
+        ```
+
+    -   `object`: If `type` is `object`, then there MAY be any of the following
+        fields at the same level as `type`: `additionalProperties`,
+        `properties`.
+        Objects are defined as sets of key/value pairs.
+        Keys MUST be strings, while values may have specific attributes,
+        which is what `additionalProperties` describes.
+        Here is an example of a field which MUST be an object,
+        with one optional field named "OptionalField" that MUST be a `string`
+        and any number of additional optional fields that MUST be `number`s.
+        ```yaml
+        name: ExampleField
+        description: |
+            The description of "ExampleField".
+        type: object
+        properties:
+            OptionalField:
+                type: string
+        additionalProperties:
+            type: number
+        ```
+
+    -   `boolean`: If `type` is `boolean`, then the field's value MUST be one
+        of: `"true"`, `"false"`. Here is an example:
```

Comment:
we want booleans to be JSON booleans, not strings of booleans, right?

```suggestion
        of: `true`, `false`. Here is an example:
```

Reply:
Good point. Addressed in 890b364.